### PR TITLE
Initial QLC+ buildbot config

### PR DESCRIPTION
**Pull request by peternewman**

### CODE REVIEW

File: build.config [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# IRC bot base name
+IRCBOTBASENAME = "QLCplus-buildbot"
+# Main IRC channel
+# Bot IRC channel
+#IRCBOTCHANNEL = "#qlcplus-build"
+
+# The port the slaves connect to
+SLAVE_PORT = 9989
+
+CPP_LINT_ARGS = '--filter=-legal/copyright,-readability/streams,-runtime/arrays'
```
Comment (nomis52) [inline build.config]: Do we want linting? Probably not unless the codebase is clean.

File: master.cfg [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+   haltOnFailure=False,
+   flunkOnFailure=False,
+   warnOnWarnings=True,
+   warnOnFailure=True,
+   description='linting',
+   descriptionDone='lint',
+))
+
+# jslint factory
+js_lint_command = (
```
Comment (nomis52) [inline master.cfg]: No Javascript from what I know.

File: master.cfg [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+   command=js_lint_command,
+   haltOnFailure=False,
+   flunkOnFailure=False,
+   warnOnWarnings=True,
+   warnOnFailure=True,
+   description='linting',
+   descriptionDone='lint',
+))
+
+# heap check factory
```
Comment (nomis52) [inline master.cfg:125]: Same for heap checking. I'd leave this out for now

File: master.cfg [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+    descriptionDone='test',
+    command=['./unittest.sh'],
+    env={'HEAPCHECK': 'normal',
+         'PPROF_PATH': '/usr/bin/google-pprof'},
+    logfiles=config['LOGFILES'],
+    lazylogfiles=True,
+))
+
+# doxygen doc factory
+doc_factory = BuildFactory()
```
Comment (nomis52) [inline master.cfg:151]: And doxygen.

File: build.config [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+# Title, 'project' is appended to the title.
+TITLE = "QLC+"
+# URL of the project
+URL = "http://qlcplus.sourceforge.net/index.shtml"
+# URL of the buildbot
+BUILDBOTURL = "https://qlcplusbuildbot.openlighting.org/"
+
+# IRC server
+IRCSERVER = "irc.freenode.org"
```
Comment (nomis52) [inline build.config:19]: I'd remove all the IRC bits.

File: build.config (hunk 1 above)
Comment (peternewman) [inline build.config]: I've commented it out and tweaked master.cfg so it skips it if the option isn't present.

File: master.cfg (hunk 2 above)
Comment (peternewman) [inline master.cfg]: Likewise. There is actually a bit, but I guess it's not closure.

File: master.cfg (hunk 3 above)
Comment (peternewman) [inline master.cfg:125]: Done. I've had to comment this out in master.cfg to not do anything odd in build.config.

File: master.cfg (hunk 4 above)
Comment (peternewman) [inline master.cfg:151]: This is now skipped due to having no configured slaves. They do have doxygen already.

File: build.config (hunk 5 above)
Comment (peternewman) [inline build.config:19]: Again they get skipped now if the config isn't present.

**Comment by nomis52** [comment]
LGTM

**Pull request merged**